raster-foundry/raster-foundry
# Rename Packages

Still a WIP as I figure out all the comprehensive testing needed and get a PR up in our deployment repository.

## Overview

Renames packages and assembly jars for more consistent naming conventions across the project and artifact publishing.

I used the following `find`/`sed` commands to do most of the edits:
```
find ./ -type f -name '*.sbt' -exec sed -i 's/com\.azavea\.rf/com\.rasterfoundry/g' {} \;
find ./ -type f -name '*.scala' -exec sed -i 's/com\.azavea\.rf/com\.rasterfoundry/g' {} \;
find ./ -type f -name '*.py' -exec sed -i 's/com\.azavea\.rf/com\.rasterfoundry/g' {} \;
find ./ -type f -name '*.py' -exec sed -i 's/rf-batch\.jar/batch-assembly\.jar/g' {} \;
```

After that I manually adjusted the directory structure of a few subprojects (`db`, `datamodel`, `bridge`, `batch`) to remove unnecessary directories that reflected the old naming convention.

I also adjusted the names of the built artifacts and synchronized the names of the subprojects within their respective `build.sbt` files.

### Checklist


## Testing Instructions

Testing end-to-end on everything is going to be pretty difficult. I imagine some of this is going to have to shake out in staging if we missing something. However, the following commands go through some basic functionality.

 - Run `./scripts/cibuild`
 - Run the following commands (one at a time) to verify that assembly jar can be accessed in batch processes (note: they will all fail, but we're just checking they can start the jar CLI):
 ```
./scripts/console batch bash
rf export abc
rf process-upload abc
rf ingest-scene 2fdc678b-80a9-4dfe-99dc-981f501a3e88
# ctrl-c the above once the ingest starts
rf find-aoi-projects abc
rf update-aoi-project abc
 ```
 - Search for `.jar` in project, ensure that all references to assembly jars have been updated to the format `<root>-assembly.jar`
 - Ensure it is possible to build containers for api and tile server:
 ```
 docker-compose run --rm --no-deps api-server api/assembly
 docker-compose run --rm --no-deps api-server tile/assembly

 docker-compose \
  -f "docker-compose.yml" \
  -f "docker-compose.test.yml"\
   build api-server tile-server
 ```
 - Ensure it is possible to start servers (though they will error, but that's ok)
```
docker run docker run raster-foundry-tile-server:latest
docker run docker run raster-foundry-api-server:latest
```
Closes #3791
